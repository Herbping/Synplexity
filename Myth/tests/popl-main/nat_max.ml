type nat =
  | Z
  | S of nat

type cmp =
  | LT
  | EQ
  | GT

let rec compare (n1 : nat) (n2 :nat) : cmp =
  match n1 with
  | Z u -> ( match n2 with 
           | Z u -> EQ
           | S m -> LT
         )
  | S m1 -> 
      ( match n2 with
      | Z u -> GT
      | S m2 -> (compare m1 m2) )
;;

let nat_max : nat -> nat -> nat |>
 /\(0 -> /\(0 -> 0, 1 -> 1, 2 -> 2),
    1 -> /\(0 -> 1, 1 -> 1, 2 -> 2),
    2 -> /\(0 -> 2, 1 -> 2, 2 -> 2)) =?
