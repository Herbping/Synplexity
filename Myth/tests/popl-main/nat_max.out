(* Generated Thu Oct  8 20:40:44 2015 *)
fun (n1:nat) ->
  fun (n2:nat) ->
    match (compare n2) n1 with
    | LT u1 -> n1
    | EQ u1 -> n2
    | GT u1 -> n2
